### PR TITLE
Cu 86cv4e9a3 onboarding api

### PR DESCRIPTION
**Pull request by anal-anckr**
pr for : https://app.clickup.com/t/86cv4e9a3
 
 changes: 
 1. truevault folder & inside that files created which contain the true vault crud functions.
Screenshot from 2024-04-22 16-06-41

 2. onboarding api is created.
 3. send email api is created. (need to be develop further)
 4. middleware file chnages takes place
 5. responce file changes takes place

### CODE REVIEW

File: src/app/responce.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ interface IErrorStack{
     message:string,
```
Comment (akshayAnckr) [inline src/app/responce.ts:5]: @anal-anckr file name typo issue : it should be 'response.ts'

**Review by akshayAnckr — CHANGES_REQUESTED** [review]

**Review by vishal-anckr — CHANGES_REQUESTED** [review]
Remove console from PR

**Comment by vishal-anckr** [comment]
@anal-anckr All checks failed. Please fix

File: src/app/api/orgReferral/create/route.ts [hunk 2]
```diff
@@ -3,37 +3,37 @@
...
 
 export async function POST(request: NextRequest, response: NextResponse) {
   const req = await request.json();
     const {orgReferralData} = req;
-    // console.log("prgData",orgReferralData)
+    console.log("prgData",orgReferralData)
   try {
 
-    if (orgReferralData) {
+    // if (orgReferralData) {
```
Comment (vishal-anckr) [inline src/app/api/orgReferral/create/route.ts:14]: @anal-anckr Commented code should be removed

File: src/app/api/orgReferral/create/route.ts [hunk 3]
```diff
@@ -3,37 +3,37 @@
...
 
-      const checkOrgCodeExistance= await checkOrgCodeExist(orgReferralData.organization_code);
-      if(checkOrgCodeExistance){
-        return failResponse('organization_code already existed', 400)
+    //   const checkOrgCodeExistance= await checkOrgCodeExist(orgReferralData.organization_code);
+    //   if(checkOrgCodeExistance){
```
Comment (vishal-anckr) [inline src/app/api/orgReferral/create/route.ts:17]: @anal-anckr Commented code should be removed

File: src/app/truevault/groups/adduser-group.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+
+    const data = await response.json();
+    return { data: data.result, error: null }
+  } catch (error) {
+    // console.error('Error adding group members:', error);
```
Comment (vishal-anckr) [inline src/app/truevault/groups/adduser-group.ts:24]: @anal-anckr Commented code should be removed

File: src/app/truevault/groups/create-group.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+
+    const data = await response.json();
+    // console.log("groupData", data)
+    return { data: data.group.group_id, error: null }
+  } catch (error) {
+    // console.error('Error creating group:', error);
```
Comment (vishal-anckr) [inline src/app/truevault/groups/create-group.ts:27]: @anal-anckr Commented code should be removed

File: src/app/truevault/groups/delete-group.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+          'Authorization': `Basic ${Buffer.from(process.env.NEXT_PUBLIC_TVDB_API_KEY + ':').toString('base64')}`
+        }
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      return true
+    } catch (error) {
+      // console.error('Error deleting group:', error);
```
Comment (vishal-anckr) [inline src/app/truevault/groups/delete-group.ts:18]: @anal-anckr Commented code should be removed

File: src/app/truevault/groups/read-group.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+
+    const data = await response.json();
+    return data
+  } catch (error) {
+    // console.error('Error retrieving group:', error);
```
Comment (vishal-anckr) [inline src/app/truevault/groups/read-group.ts:19]: @anal-anckr Commented code should be removed

File: src/app/truevault/groups/update-group.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+
+    const data = await response.json();
+    return data
+  } catch (error) {
+    // console.error('Error updating group:', error);
```
Comment (vishal-anckr) [inline src/app/truevault/groups/update-group.ts:22]: @anal-anckr Commented code should be removed

**Review by vishal-anckr — CHANGES_REQUESTED** [review]
@anal-anckr Few mistakes we need to improve:
- Unused commented code
- Inconsistency in string response message.

File: src/app/api/orgReferral/create/route.ts (hunk 2 above)
Comment (anal-anckr) [reply src/app/api/orgReferral/create/route.ts:14]: no that code has to be there, for future reference

Comment (anal-anckr) [reply src/app/api/orgReferral/create/route.ts:14]: after sprint /milestone 4 ,after knowing the admin panel scope( if any) i will delete that code

**Pull request merged**